integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/projection/IdentifierSearchProjectionSingleValuedIT.java

```diff
@@ -0,0 +1,205 @@
+			.name( "my-index" );
+
+	private final SimpleMappedIndex<IndexBinding> compatibleIndex = SimpleMappedIndex.of( IndexBinding::new )
+			.name( "my-compatible-index" );
+
+	private final StubMappedIndex incompatibleIndex = StubMappedIndex.ofAdvancedNonRetrievable( ctx -> ctx
+			.idDslConverter( INCOMPATIBLE_ID_CONVERTER ) );
+
+	private final String[] ids = new String[7];
+	private final String[] names = new String[7];
+	private final List<String>[] duplicates = new List[7];
+
+	public IdentifierSearchProjectionSingleValuedIT() {
+		initValues();
+	}
+
+	@Before
+	public void setup() {
+		setupHelper.start().withIndexes( index, compatibleIndex, incompatibleIndex ).setup();
+		initData();
+	}
+
+	@Test
+	public void simple() {
+		StubMappingScope scope = index.createScope();
+		SearchQuery<String> query = scope.query()
+				.select( f -> f.id( String.class ) )
+				.where( f -> f.matchAll() )
+				.toQuery();
+
+		List<String> result = query.fetchHits( 30 );
+		assertThat( result ).containsExactlyInAnyOrder( ids );
+	}
+
+	@Test
+	public void noClass() {
+		StubMappingScope scope = index.createScope();
+		SearchQuery<Object> query = scope.query()
+				.select( f -> f.id() )
+				.where( f -> f.matchAll() )
+				.toQuery();
+
+		List<Object> result = query.fetchHits( 30 );
+		assertThat( result ).containsExactlyInAnyOrder( (Object[]) ids );
+	}
+
+	@Test
+	public void duplicated() {
+		StubMappingScope scope = index.createScope();
+
+		SearchQuery<List<?>> query = scope.query()
+				.select( f -> f.composite( f.id( String.class ), f.id( String.class ) ) )
+				.where( f -> f.matchAll() )
+				.toQuery();
+
+		List<List<?>> result = query.fetchHits( 30 );
+		assertThat( result ).containsExactlyInAnyOrder( duplicates );
+	}
+
+	@Test
+	public void validSuperClass() {
+		StubMappingScope scope = index.createScope();
+		SearchQuery<CharSequence> query = scope.query()
+				.select( f -> f.id( CharSequence.class ) )
+				.where( f -> f.matchAll() )
+				.toQuery();
+
+		List<CharSequence> result = query.fetchHits( 30 );
+		assertThat( result ).containsExactlyInAnyOrder( ids );
+	}
+
+	@Test
+	public void invalidProjectionType() {
+		StubMappingScope scope = index.createScope();
+
+		assertThatThrownBy( () -> scope.projection().id( Integer.class ) )
+				.isInstanceOf( SearchException.class )
+				.hasMessageContainingAll(
+						"The required identifier type 'java.lang.Integer' does not match the actual identifier type 'java.lang.String'",
+						"the required identifier must be a superclass of the actual identifier"
+				);
+	}
+
+	@Test
+	public void multiIndex_withCompatibleIndex() {
+		StubMappingScope scope = index.createScope( compatibleIndex );
+		SearchQuery<String> query = scope.query()
+				.select( f -> f.id( String.class ) )
+				.where( f -> f.matchAll() )
+				.toQuery();
+
+		List<String> result = query.fetchHits( 30 );
+		assertThat( result ).containsExactlyInAnyOrder( ids );
```

Comment:
This test would make more sense if the compatible index actually contained documents, don't you think?